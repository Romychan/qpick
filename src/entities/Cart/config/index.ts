/** The name of the storage for storing data in the cart */
export const CART_STORAGE_NAME = 'cart';
