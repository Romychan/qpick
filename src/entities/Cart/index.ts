export { MOCK_CART, MOCK_CART_PRODUCT } from './lib/mock';

export { CartSummary } from './ui/CartSummary/CartSummary';

export { CartContext, CartContextProvider } from './model/context';
export { useCartStore } from './model/hooks';
