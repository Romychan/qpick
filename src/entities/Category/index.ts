export type { Category } from './model/types';
export { MAIN_CATEGORIES } from './model/constants';
