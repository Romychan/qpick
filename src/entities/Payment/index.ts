export { PAYMENT_TYPES } from './model/constants';
export {
  PaymentFormSchema,
  CardPaymentSchema,
  CashPaymentSchema,
} from './model/schemas';
export type {
  PaymentFormType,
  CardPaymentFormType,
  CashPaymentFormType,
} from './model/types';
