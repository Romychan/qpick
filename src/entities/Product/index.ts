export { ProductDetails } from './ui/ProductDetails/ProductDetails';
export { ProductRow } from './ui/ProductRow/ProductRow';
export { ProductCard } from './ui/ProductCard/ProductCard';

export type { Product } from './model/types';

export { MOCK_PRODUCTS } from './lib/mock';
