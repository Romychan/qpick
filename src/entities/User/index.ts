export { UserFormSchema } from './model/schemas';
export type { UserFormType } from './model/types';
