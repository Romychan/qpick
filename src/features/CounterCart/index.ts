export { CounterCart } from './ui/CounterCart';
