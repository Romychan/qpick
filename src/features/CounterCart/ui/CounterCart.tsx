import { useCartStore } from '~/entities/Cart';

import { Counter } from '~/shared/ui/Counter';
import { formatPrice } from '~/shared/lib/utils';

import styles from './CounterCart.module.scss';

interface ICounterCartProps {
  /** The ID of the product to change the quantity in the cart */
  id: number;
  /** The default price of the item in the list */
  price: number;
  /** The quantity of product in the cart */
  quantity: number;
}

/** This component is used to display the current value and change the number of products in the cart */
export const CounterCart = ({ id, price, quantity }: ICounterCartProps) => {
  const { increaseQuantity, decreaseQuantity } = useCartStore();

  return (
    <div className={styles.container}>
      <Counter
        count={quantity}
        onIncrement={() => increaseQuantity({ id })}
        onDecrement={() => decreaseQuantity({ id })}
      />

      <p className={styles.total} data-testid="cart-counter-total">
        {formatPrice(price * quantity)}
      </p>
    </div>
  );
};
