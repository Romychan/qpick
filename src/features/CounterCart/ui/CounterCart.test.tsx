import { fireEvent, render, screen } from '@testing-library/react';
import { vi } from 'vitest';

import { MOCK_CART } from '~/entities/Cart';
import { CartContext } from '~/entities/Cart';

import { CounterCart } from './CounterCart';

const MOCK_CART_STORE = {
  items: MOCK_CART,
  addToCart: vi.fn(),
  deleteFromCart: vi.fn(),
  increaseQuantity: vi.fn(),
  decreaseQuantity: vi.fn(),
  resetCart: vi.fn(),
};

describe('CounterCart', () => {
  it('should render correctly', () => {
    render(
      <CartContext.Provider value={MOCK_CART_STORE}>
        <CounterCart
          id={MOCK_CART[0].product.price}
          price={MOCK_CART[0].product.price}
          quantity={MOCK_CART[0].quantity}
        />
      </CartContext.Provider>,
    );

    const totalPriceElement = screen.getByTestId('cart-counter-total');

    expect(totalPriceElement).toBeInTheDocument();
  });

  it('should increasing the quantity of product work correctly', () => {
    const spy = vi.spyOn(MOCK_CART_STORE, 'increaseQuantity');

    render(
      <CartContext.Provider value={MOCK_CART_STORE}>
        <CounterCart
          id={MOCK_CART[0].product.price}
          price={MOCK_CART[0].product.price}
          quantity={MOCK_CART[0].quantity}
        />
      </CartContext.Provider>,
    );

    const incrementElement = screen.getByTestId('counter-increment');

    fireEvent.click(incrementElement);

    expect(spy).toHaveBeenCalled();
  });

  it('should decreasing the quantity of product work correctly', () => {
    const spy = vi.spyOn(MOCK_CART_STORE, 'decreaseQuantity');

    render(
      <CartContext.Provider value={MOCK_CART_STORE}>
        <CounterCart
          id={MOCK_CART[0].product.price}
          price={MOCK_CART[0].product.price}
          quantity={MOCK_CART[0].quantity}
        />
      </CartContext.Provider>,
    );

    const decrementElement = screen.getByTestId('counter-decrement');

    fireEvent.click(decrementElement);

    expect(spy).toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    const container = render(
      <CartContext.Provider value={MOCK_CART_STORE}>
        <CounterCart
          id={MOCK_CART[0].product.price}
          price={MOCK_CART[0].product.price}
          quantity={MOCK_CART[0].quantity}
        />
      </CartContext.Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
