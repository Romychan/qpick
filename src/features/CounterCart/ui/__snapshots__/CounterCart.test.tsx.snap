// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CounterCart > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_container_daf3ee"
      >
        <div
          class="_container_31df47"
        >
          <button
            aria-label="Уменьшение счетчика"
            class="_button_0302f3 _button_31df47 _secondary_0302f3 _sm_0302f3 _icon_0302f3"
            data-testid="counter-decrement"
            title="Уменьшить"
            type="button"
          >
            <svg
              class="icon icon-minus-filled filled"
              data-testid="icon"
              height="16"
              width="16"
            >
              <use
                xlink:href="/images/icons.svg#minus-filled"
              />
            </svg>
          </button>
          <span
            class="_counter_31df47"
            data-testid="counter-count"
          >
            2
          </span>
          <button
            aria-label="Увеличение счетчика"
            class="_button_0302f3 _button_31df47 _secondary_0302f3 _sm_0302f3 _icon_0302f3"
            data-testid="counter-increment"
            title="Увеличить"
            type="button"
          >
            <svg
              class="icon icon-plus-filled filled"
              data-testid="icon"
              height="16"
              width="16"
            >
              <use
                xlink:href="/images/icons.svg#plus-filled"
              />
            </svg>
          </button>
        </div>
        <p
          class="_total_daf3ee"
          data-testid="cart-counter-total"
        >
          2 400 ₽
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_container_daf3ee"
    >
      <div
        class="_container_31df47"
      >
        <button
          aria-label="Уменьшение счетчика"
          class="_button_0302f3 _button_31df47 _secondary_0302f3 _sm_0302f3 _icon_0302f3"
          data-testid="counter-decrement"
          title="Уменьшить"
          type="button"
        >
          <svg
            class="icon icon-minus-filled filled"
            data-testid="icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="/images/icons.svg#minus-filled"
            />
          </svg>
        </button>
        <span
          class="_counter_31df47"
          data-testid="counter-count"
        >
          2
        </span>
        <button
          aria-label="Увеличение счетчика"
          class="_button_0302f3 _button_31df47 _secondary_0302f3 _sm_0302f3 _icon_0302f3"
          data-testid="counter-increment"
          title="Увеличить"
          type="button"
        >
          <svg
            class="icon icon-plus-filled filled"
            data-testid="icon"
            height="16"
            width="16"
          >
            <use
              xlink:href="/images/icons.svg#plus-filled"
            />
          </svg>
        </button>
      </div>
      <p
        class="_total_daf3ee"
        data-testid="cart-counter-total"
      >
        2 400 ₽
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
